x-pack/plugins/file_upload/public/components/json_upload_and_parse.tsx

```diff
@@ -274,7 +274,11 @@ export class JsonUploadAndParse extends Component<FileUploadComponentProps, Stat
     });
 
     const isReadyToImport = !!name && error === undefined;
-    this.props.onIndexReady(isReadyToImport);
+    if (isReadyToImport) {
+      this.props.enableImportBtn();
+    } else {
+      this.props.disableImportBtn();
+    }
```

Comment:
Why move this from the wizard in Maps? Seems like whether a button hosted in the Maps app is enabled or not should stay a Maps concern

Reply:
> Seems like whether a button hosted in the Maps app is enabled or not should stay a Maps concern

The component needs to be able to enable/disable the import button. For example, when the index name is getting validated, the component needs to be able to disable the index name until validation has completed.

Reply:
No real issue with having a callback, I would just make it more generic to fit with the others and not be specific to how Maps uses it. Something like `onImportReady` and `onImportNotReady`. Your call.